contracts/RFQv2.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+import { TokenCollector } from "./utils/TokenCollector.sol";
+import { BaseLibEIP712 } from "./utils/BaseLibEIP712.sol";
+import { Asset } from "./utils/Asset.sol";
+import { Offer, getOfferHash } from "./utils/Offer.sol";
+import { RFQOrder, getRFQOrderHash } from "./utils/RFQOrder.sol";
+import { LibConstant } from "./utils/LibConstant.sol";
+import { SigCheck } from "./utils/SigCheck.sol";
+import { StrategyBase } from "./utils/StrategyBase.sol";
+import { IRFQv2 } from "./interfaces/IRFQv2.sol";
+
+/// @title RFQv2 Contract
+/// @author imToken Labs
+contract RFQv2 is IRFQv2, StrategyBase, TokenCollector, BaseLibEIP712 {
+    using SafeMath for uint256;
+    using Asset for address;
+
+    address payable public feeCollector;
+
+    /// @notice Emitted when fee collector address is updated
+    /// @param newFeeCollector The address of the new fee collector
+    event SetFeeCollector(address newFeeCollector);
+
+    receive() external payable {}
+
+    constructor(
+        address _owner,
+        address _userProxy,
+        address _weth,
+        address _permStorage,
+        address _spender,
+        address _uniswapPermit2,
+        address payable _feeCollector
+    ) StrategyBase(_owner, _userProxy, _weth, _permStorage, _spender) TokenCollector(_uniswapPermit2, _spender) {
+        feeCollector = _feeCollector;
+    }
+
+    /// @notice Set fee collector
+    /// @notice Only owner can call
+    /// @param _newFeeCollector The address of the new fee collector
+    function setFeeCollector(address payable _newFeeCollector) external onlyOwner {
+        require(_newFeeCollector != address(0), "zero address");
+        feeCollector = _newFeeCollector;
+
+        emit SetFeeCollector(_newFeeCollector);
+    }
+
+    function fillRFQ(
+        RFQOrder calldata order,
+        bytes calldata makerSignature,
+        bytes calldata makerTokenPermit,
+        bytes calldata takerTokenPermit,
+        bytes calldata takerSignature
```

Comment:
nit: How about keeping args in consistent order? For example

```solidity
// permit, sig, permit, sig
bytes calldata makerTokenPermit,
bytes calldata makerSignature,
bytes calldata takerTokenPermit,
bytes calldata takerSignature
// or permit first, then sig
bytes calldata makerTokenPermit,
bytes calldata takerTokenPermit,
bytes calldata makerSignature,
bytes calldata takerSignature
```

Reply:
I thought about it but realized that it would be inconsistent with V6 interface. Let's change v5 first since v6 is not ready.

Reply:
Didn't notice this when reviewing the v6 PR 🤣

Reply:
Adjusted in 7a11664